src/testapp/sessions/orbits.cpp

```diff
@@ -0,0 +1,232 @@
+    OSP_DECLARE_GET_DATA_IDS(uniScnFrame, TESTAPP_DATA_UNI_SCENEFRAME);
+
+    auto const tgUCore = uniCore.get_pipelines<PlUniCore>();
+
+    auto &rUniverse = top_get<Universe>(topData, idUniverse);
+
+    static constexpr int precision = 10;
+    static constexpr int planetCount = 64;
+    static constexpr int seed = 1337;
+    static constexpr spaceint_t maxDist = math::mul_2pow<spaceint_t, int>(20000ul, precision);
+    static constexpr float maxVel = 800.0f;
+
+    // Create coordinate spaces
+    CoSpaceId const mainSpace = rUniverse.m_coordIds.create();
+    std::vector<CoSpaceId> satSurfaceSpaces(planetCount);
+    rUniverse.m_coordIds.create(satSurfaceSpaces.begin(), satSurfaceSpaces.end());
+
+    rUniverse.m_coordCommon.resize(rUniverse.m_coordIds.capacity());
+
+    CoSpaceCommon &rMainSpaceCommon = rUniverse.m_coordCommon[mainSpace];
+    rMainSpaceCommon.m_satCount = planetCount;
+    rMainSpaceCommon.m_satCapacity = planetCount;
+
+    // Associate each planet satellite with their surface coordinate space
+    for (SatId satId = 0; satId < planetCount; ++satId)
+    {
+        CoSpaceId const surfaceSpaceId = satSurfaceSpaces[satId];
+        CoSpaceCommon &rCommon = rUniverse.m_coordCommon[surfaceSpaceId];
+        rCommon.m_parent = mainSpace;
+        rCommon.m_parentSat = satId;
+    }
+
+    // Coordinate space data is a single allocation partitioned to hold positions, velocities, and
+    // rotations.
+    // TODO: Alignment is needed for SIMD (not yet implemented). see Corrade alignedAlloc
+
+    std::size_t bytesUsed = 0;
+
+    // Positions and velocities are arranged as XXXX... YYYY... ZZZZ...
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satPositions[0]);
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satPositions[1]);
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satPositions[2]);
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satVelocities[0]);
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satVelocities[1]);
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satVelocities[2]);
+
+    // Rotations use XYZWXYZWXYZWXYZW...
+    partition(bytesUsed, planetCount, rMainSpaceCommon.m_satRotations[0],
+                rMainSpaceCommon.m_satRotations[1],
+                rMainSpaceCommon.m_satRotations[2],
+                rMainSpaceCommon.m_satRotations[3]);
+
+    // Allocate data for all planets
+    rMainSpaceCommon.m_data = Array<unsigned char>{Corrade::NoInit, bytesUsed};
+
+    // Create easily accessible array views for each component
+    auto const [x, y, z] = sat_views(rMainSpaceCommon.m_satPositions, rMainSpaceCommon.m_data, planetCount);
+    auto const [vx, vy, vz] = sat_views(rMainSpaceCommon.m_satVelocities, rMainSpaceCommon.m_data, planetCount);
+    auto const [qx, qy, qz, qw] = sat_views(rMainSpaceCommon.m_satRotations, rMainSpaceCommon.m_data, planetCount);
+
+    std::mt19937 gen(seed);
```

Comment:
you might get slightly better random number generation performance with the _64 version of the mt19937